src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java

```diff
@@ -134,6 +135,34 @@ public static User parse(XContentParser parser) throws IOException {
         return new User(name, backendRoles, roles, customAttNames);
     }
 
+    /**
+     * User String format must be pipe separated as : user_name|backendrole1,backendrole2|roles1,role2
+     * @param userString
+     * @return
+     */
+    public static User parse(final String userString) {
+        if (Strings.isNullOrEmpty(userString)) {
+            return null;
+        }
+
+        String[] strs = userString.split("\\|");
+        if ((strs.length == 0) || (Strings.isNullOrEmpty(strs[0]))) {
+            return null;
+        }
+
+        String userName = strs[0].trim();
+        List<String> backendRoles = new ArrayList<>();
+        List<String> roles = new ArrayList<>();
+
+        if ((strs.length > 1) && !Strings.isNullOrEmpty(strs[1])) {
+            backendRoles.addAll(Arrays.asList(strs[1].split(",")));
+        }
+        if ((strs.length > 2) && !Strings.isNullOrEmpty(strs[2])) {
+            roles.addAll(Arrays.asList(strs[2].split(",")));
+        }
+        return new User(userName, backendRoles, roles, Arrays.asList());
```

Comment:
Is it by design to not support custom attribute names?

Reply:
Yes. And also we want to keep injected string light weight. Custom attributes could make the string large.